### PR TITLE
Set Database Collation/charset to GB

### PR DESCRIPTION
**Pull request by kunlebalogun**
### Change description ###
Need to set PostgreSQL  FlexServer DB collation / charset to en_GB.utf8

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

**Review by thomast1906 — CHANGES_REQUESTED** [review]
as discussed, update `null` to a default value

**Review by TomKimber — APPROVED** [review]

File: inputs-optional.tf [hunk 1]
```diff
@@ -84,26 +84,20 @@ variable "enable_read_only_group_access" {
...
+  default     = "utf8"
+  description = "Specifies the Charset for the Azure PostgreSQL Flexible Server Database, which needs to be a valid PostgreSQL Charset."
 }
 
-variable "add_multiple_readonly_groups" {
+variable "high_availability" {
   type        = bool
   default     = false
-  description = "Allows to add multipe AAD read only group for the postgres DB"
+  description = "Enable high availability mode for the PostgreSQL Flexible Server."
```
Comment (timja) [inline inputs-optional.tf:102]: I think this should have a readme update should make it clear this needs to be enabled in prod and probably aat/staging and perftest/test

File: inputs-optional.tf [hunk 2]
```diff
@@ -84,26 +84,20 @@ variable "enable_read_only_group_access" {
...
+variable "charset" {
   type        = string
-  default     = null
-  description = "Comma seperated additional admin groups"
+  default     = "utf8"
+  description = "Specifies the Charset for the Azure PostgreSQL Flexible Server Database, which needs to be a valid PostgreSQL Charset."
 }
 
-variable "add_multiple_readonly_groups" {
+variable "high_availability" {
```
Comment (timja) [inline inputs-optional.tf:99]: this doesn't appear to be used any more

File: inputs-optional.tf (hunk 1 above)
Comment (timja) [reply inputs-optional.tf:102]:
```suggestion
  description = "Overrides the automatic selection of high availability mode for the PostgreSQL Flexible Server. Generally you shouldn't set this yourself."
```

File: pgsql-flexible-server.tf [hunk 3]
```diff
@@ -9,9 +9,15 @@ locals {
...
-  db_reader_user = local.is_prod ? (var.add_multiple_readonly_groups ? split(",", join(",", ["DTS JIT Access ${var.product} DB Reader SC", var.additional_readonly_groups])) : split(",", "DTS JIT Access ${var.product} DB Reader SC")) : (var.add_multiple_readonly_groups ? split(",", join(",", ["DTS ${upper(var.business_area)} DB Access Reader", var.additional_readonly_groups])) : split(",", "DTS ${upper(var.business_area)} DB Access Reader"))
+  is_prod = length(regexall(".*(prod).*", var.env)) > 0
+
+  admin_group    = local.is_prod ? "DTS Platform Operations SC" : "DTS Platform Operations"
+  db_reader_user = local.is_prod ? "DTS JIT Access ${var.product} DB Reader SC" : "DTS ${upper(var.business_area)} DB Access Reader"
+
+
+  high_availability = var.high_availability == true || var.env == "prod" || var.env == "ptl" || var.env == "perftest" || var.env == "stg" || var.env == "aat" ? true : false
```
Comment (timja) [inline pgsql-flexible-server.tf:18]:
```suggestion
  high_availability_environments = ["ptl", "perftest", "stg", "aat", "prod"]
  high_availability = var.high_availability == true || contains(local.high_availability_environments, var.env)
```

File: pgsql-flexible-server.tf [hunk 4]
```diff
@@ -130,12 +137,12 @@ resource "azurerm_postgresql_flexible_server_active_directory_administrator" "pg
 }
 
 resource "null_resource" "set-user-permissions-additionaldbs" {
-  count = var.enable_read_only_group_access ? length(local.db_reader_user) : 0
+  for_each = var.enable_read_only_group_access ? { for index, db in var.pgsql_databases : db.name => db } : {}
```
Comment (thomast1906) [inline pgsql-flexible-server.tf:140]: @cpareek could it be reviewed to confirm that is the simplest way to write this?

Comment (cpareek) [reply pgsql-flexible-server.tf:141]: sure - I have checked with Tim on this too as we wanted to only run the block if the enable_read_only_group_access is true, the for_each is taken from existing code here..
https://github.com/hmcts/terraform-module-postgresql-flexible/blob/b5cfd03383426c5eb4a6784773ab9ea4a94100f8/pgsql-flexible-server-db.tf#L2

File: pgsql-flexible-server.tf [hunk 5]
```diff
@@ -144,8 +152,8 @@ resource "null_resource" "set-user-permissions-additionaldbs" {
     environment = {
       DB_HOST_NAME   = azurerm_postgresql_flexible_server.pgsql_server.fqdn
       DB_USER        = "${data.azuread_service_principal.mi_name[0].display_name}"
```
Comment (timja) [inline pgsql-flexible-server.tf:154]:
```suggestion
      DB_USER        = data.azuread_service_principal.mi_name[0].display_name
```

File: set-postgres-permissions.bash [hunk 6]
```diff
@@ -1,12 +1,12 @@
-#!/usr/bin/env bash
+#!/usr/bin/env bash
```
Comment (timja) [inline set-postgres-permissions.bash:1]:
```suggestion
#!/usr/bin/env bash
```

File: set-postgres-permissions.bash [hunk 7]
```diff
@@ -20,9 +20,14 @@ BEGIN
 END
 \$do\$;
 
+"
+
+SQL_COMMAND="
```
Comment (timja) [inline set-postgres-permissions.bash:25]: please move this down to above the psql call that it's being used for, that will make it clearer why there's two different scripts

File: pgsql-flexible-server.tf (hunk 5 above)
Comment (cpareek) [reply pgsql-flexible-server.tf:154]: sure - changed now

File: set-postgres-permissions.bash (hunk 7 above)
Comment (cpareek) [reply set-postgres-permissions.bash:25]: sure - done

File: set-postgres-permissions.bash (hunk 6 above)
Comment (cpareek) [reply set-postgres-permissions.bash:1]: updated

**Review by timja — APPROVED** [review]

**Review by thomast1906 — APPROVED** [review]

**Pull request merged**